content/core-features/presence-and-occupancy/topic-presence_ia-update.textile

```diff
@@ -0,0 +1,737 @@
+  options.clientId = "bob"
+  let realtime = ARTRealtime(options: options)
+  let channel = realtime.channels.get("{{RANDOM_CHANNEL_NAME}}")
+  channel.presence.subscribe(.enter) { member in
+      print("Member \(member.clientId) entered")
+  }
+  channel.presence.enter(nil)
+```
+
+<span lang="ruby">Note that all examples above assume you are running them within an EventMachine reactor. Find out more in our "Realtime usage documentation":/realtime/usage.</span>
+
+If you would prefer to just dive into code and see some examples of how to use presence via the Realtime API, then we recommend you take a look at our "Realtime tutorials":/tutorials.
+
+h3(#getting-started-rest). Getting started with the REST API
+
+Using the "Ably REST API":/rest-api it is possible to obtain the set of members currently present on a channel, or obtain the presence history for the channel, if persistence is enabled for that channel.
+
+Since the "Ably REST API":/rest-api is stateless, and REST clients do not have realtime connections to the Ably service, it is not possible to enter or leave a channel via the REST API. Find out more about "using presence with the Realtime API":/realtime/presence.
+
+A single <span lang="default">"@clientId@":/api/rest-sdk#client-options</span><span lang="ruby,python">"@client_id@":/api/rest-sdk#client-options</span><span lang="csharp">"@ClientId@":/api/rest-sdk#client-options</span> may be present multiple times on the same channel via different client connections. As far as Ably is concerned, these are different members of the presence set for the channel, however they will be differentiated by their unique "<span lang="default">@connectionId@</span><span lang="csharp">@ConnectionId@</span><span lang="ruby">@connection_id@</span>":/api/realtime-sdk/connection#id. For example, if a client with ID "Sarah" is connected to a chat channel on both a desktop and a mobile device simultaneously, "Sarah" will be present twice in the presence member set with the same client ID, yet will have two unique connection IDs. A member of the presence set is therefore unique by the combination of the <span lang="default">"@clientId@":/api/rest-sdk#client-options</span><span lang="ruby,python">"@client_id@":/api/rest-sdk#client-options</span><span lang="csharp">"@ClientId@":/api/rest-sdk#client-options</span> and "<span lang="default">@connectionId@</span><span lang="csharp">@ConnectionId@</span><span lang="ruby">@connection_id@</span>":/api/realtime-sdk/connection#id strings.
+
+If you would prefer to just dive into code and see some examples of how to use presence via the REST API, then we recommend you take a look at our "REST tutorials":/tutorials.
+
+h2(#presence-subscribing-states). Subscribing to states and events
+
+<%= partial partial_version('shared/_presence_states') %>
+
+"View a presence states and events example":<%= JsBins.url_for('realtime/presence-events') %>
+
+h3(#member-data). Member data
+
+h4(#realtime-member-data). Realtime API
```

Comment:
Aren't we in the REST getting started? Was surprised to see this "Realtime API" section here.